src/service/redirectToOriginalUrl.js

```diff
@@ -0,0 +1,21 @@
+function redirectToOriginalUrl(shortCode, urlDatabase) {
+    try {
+        const urlEntry = urlDatabase[shortCode];
+
+        if (!urlEntry) {
+            return {
+                status: 'error',
+                message: 'Short URL not found'
+            };
+        } else {
+            return {
+                status: 'success',
+                redirect: urlEntry.originalUrl
+            };
+        }
+    } catch (error) {
+        throw new Error('Internal server error');
+    }
+}

```

Comment:
Esse resgate de erro no service não é tão necessário, ele só suprime a mensagem original e lança um Error genérico para o controller. Como você já tá resgatando e tratando no controller, não vejo problema em remover esse try/catch por inteiro aqui.
```suggestion
        const urlEntry = urlDatabase[shortCode];

        if (!urlEntry) {
            return {
                status: 'error',
                message: 'Short URL not found'
            };
        } else {
            return {
                status: 'success',
                redirect: urlEntry.originalUrl
            };
        }
}

```

Reply:
Além disso, não consegui encontrar alguma instrução nessa função que poderia lançar um erro. Como não há nenhum throw previsto, geralmente não é interessante você suprimir erros que você não espera